crates/tags/src/storage/memory.rs

```diff
@@ -29,18 +29,25 @@ impl Storage for Memory {
     type Error = StatusCode;
 
     async fn tags(&self) -> Result<Vec<Tag>, Self::Error> {
-        todo!()
+        let our_map = self.0.read().await;
+        Ok(our_map.keys().cloned().collect())
```

Comment:
```suggestion
        let lock = self.0.read().await;
        Ok(lock.keys().cloned().collect())
```

The reason for this rename is that `lock` is actually your handle on the lock. It isn't a map. It **dereferences** to a map. But it isn't actually a map itself.

Reply:
Do this same rename everywhere.